src/smefit/optimize/analytic.py

```diff
@@ -147,6 +149,31 @@ def run_sampling(self):
             @ diff_sm
         )
 
+        # Compute some metrics of the fit result
+        coeffs_name = sorted(self.coefficients.free_parameters.index)
+
+        fit_result["best_fit_point"] = dict(zip(coeffs_name, coeff_best))
+
+        self.coefficients.set_free_parameters(coeff_best)
+        self.coefficients.set_constraints()
+
+        # compute max log likelihood
+        chi2_tot = chi2.compute_chi2(
+            self.loaded_datasets,
+            self.coefficients.value,
+            self.use_quad,
+            self.use_multiplicative_prescription,
+        )
+        max_logl = float(-0.5 * chi2_tot)
+
+        fit_result["max_log_likelihood"] = max_logl
+
+        gaussian_integral = np.log(np.sqrt(np.linalg.det(2 * np.pi * coeff_covmat)))
+        # NOTE: current formula for logz is not inluding the prior penalty
+        logz = gaussian_integral + max_logl
```

Comment:
Here we assume the prior is a delta at the best fit point, don't we? That's what you mean by excluding the prior penalty if I understand correctly.

Reply:
The log Z (approximately) is given by something like
gaussian_integral + max_logl + log_prior, 
where log_prior is the log of the volume of the hypercube: (a1 -b1) * (a2 -b2) * ..... * (an - bn), where (ai, bi) is the prior of the Ci wilson coefficient. It's doable by using the max and min specified in the runcard, but then one needs to make sure that they are large enough otherwise the approximation is not valid etc etc. I thought it was an unnecessary implementation at the moment, especially given that we have it working in ultranest.